[readme] Add 'ruff', install at the top, paragraph title to permit skimming (#8194)
PyCQA/pylint: README.rst

Pylint analyses your code without actually running it. It checks for errors, enforces a
coding standard, looks for `code smells`_, and can make suggestions about how the code
could be refactored.

.. _`code smells`: https://martinfowler.com/bliki/CodeSmell.html

Install
-------

.. This is used inside the doc to recover the start of the short text for installation

For command line use, pylint is installed with::

    pip install pylint

Or if you want to also check spelling with ``enchant`` (you might need to
`install the enchant C library <https://pyenchant.github.io/pyenchant/install.html#installing-the-enchant-c-library>`_):

.. code-block:: sh

   pip install pylint[spelling]

It can also be integrated in most editors or IDEs. More information can be found
`in the documentation`_.

.. _in the documentation: https://pylint.readthedocs.io/en/latest/user_guide/installation/index.html

.. This is used inside the doc to recover the end of the short text for installation

What differentiate Pylint?

> --------------------------

Author: ```suggestion
---------------------------
```

Pylint is not trusting your typing and is inferring the actual value of nodes (for a
start because there was no typing when pylint started off) using its internal code
representation (astroid). If your code is ``import logging as argparse``, Pylint

> can check and know that ``argparse.error(...)`` is in fact a logging call and not an

Reviewer: ```suggestion
can check and knows that ``argparse.error(...)`` is in fact a logging call and not an
```

argparse call. This  makes pylint slower, but it also let pylint find more issues if
your code is not fully typed.

    [inference] is the killer feature that keeps us using [pylint] in our project despite how painfully slow it is.
    - `Realist pylint user`_, 2022

.. _`Realist pylint user`: https://github.com/charliermarsh/ruff/issues/970#issuecomment-1381067064

pylint, not afraid of being a little slower than it already is, is also a lot more thorough than other linters.

> There is more checks, some opinionated one that are deactivated by default, but can be enabled using configuration.

Reviewer: ```suggestion
There are more checks, some opinionated ones that are deactivated by default, but can be enabled using configuration.
```
